### PR TITLE
[HWKMETRICS-591] Incorrect token authorization takes longer than correct..

### PR DESCRIPTION
**Pull request by burmanm**
Do not retry token authorization if connection was successful, but the response was not what we wanted.

@mwringe

### CODE REVIEW

File: containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/TokenAuthenticator.java [hunk 1]
```diff
@@ -490,17 +487,18 @@ protected void stringDone(String body) {
...
                     onRequestResult(serverExchange, connection, allowed);
                 } catch (IOException e) {
-                    onRequestFailure(serverExchange, connection, e);
+                    onRequestFailure(serverExchange, connection, e, true);
                 }
             } else {
                 IOException e = new IOException(StatusCodes.getReason(clientExchange.getResponse().getResponseCode()));
-                onRequestFailure(serverExchange, connection, e);
+                // Retries will not help to solve the issue
+                onRequestFailure(serverExchange, connection, e, false);
```
Comment (mwringe) [inline containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/TokenAuthenticator.java]: This gets call if the response is anything other than 201. And there are some responses in which we may want to retry (eg 503) and certain ones in which we don't want to retry (eg 400, 401, 403, etc).

I think it makes sense to still retry in some situations which would need to be addressed here.

Comment (burmanm) [inline containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/TokenAuthenticator.java]: I tried to search the Kubernetes documentation on which return codes are available, but the documentation in this sense is useless. 503 is service unavailable, transforming it to 500 sounds like a proper choice in my opinion? It's not necessarily quickly available again.

Comment (mwringe) [inline containers/hawkular-openshift-security-filter/src/main/java/org/hawkular/openshift/auth/TokenAuthenticator.java]: I don't think we should worry too much about what the current kubernetes implementation is doing, we might not even be hitting the exact endpoint but some proxy or a load balancer.

I think this should just be done with best effort and based on the definitions of what those status codes typically means (eg 503 typically is a temporary problem, but not always). Even if we just retry with a 503 and not retry with the others I think would be a good solution.

If we have a need to tweak this later if deemed necessary.

**Comment by mwringe** [comment]
Awesome, LGTM

**Comment by stefannegrea** [comment]
Thanks @burmanm!

**Pull request merged**